js/__internal/grids/grid_core/filter_panel/module.ts

```diff
@@ -41,14 +43,18 @@ const FilterPanelView = modules.View.inherit({
       .addClass(that.addWidgetPrefix(FILTER_PANEL_LEFT_CONTAINER))
       .appendTo($element);
 
-    if (that.option('filterValue') || that._filterValueBuffer) {
-      $leftContainer.append(that._getCheckElement())
-        .append(that._getFilterElement())
-        .append(that._getTextElement());
-      $element.append(that._getRemoveButtonElement());
-    } else {
-      $leftContainer.append(that._getFilterElement())
-        .append(that._getTextElement());
+    const isColumnsDefined = !!this._columnsController.getColumns().length;
+
+    if (isColumnsDefined) {
```

Comment:
below text of filter builder is generated. So for generating it we need only column.name and we can not check column types via columnsController.isAllDataTypesDefined()

Reply:
Let's move this code to a separate render function

Something like this:
```typescript
  _renderCore(): void {
     // other code.

     // Is it safe to call getColumns().length here?
     const isColumnsDefined = !!this._columnsController.getColumns().length;

    if (isColumnsDefined) {
      this._renderFilterBuilderText($element, $leftContainer);
    }
  }

  _renderFilterBuilderText(
    $element: dxElementWrapper,
     $leftContainer: dxElementWrapper
  ): void {
    const $filterElement = this._getFilterElement();
    const $textElement = this._getTextElement();
    
    if(!this.option('filterValue') && !this._filterValueBuffer) {
      $leftContainer
        .append($filterElement)
        .append($textElement);
      return;
    } 
    
    const $checkElement = this._getCheckElement();
    const $removeButtonElement = this._getRemoveButtonElement();

    $leftContainer
      .append(this._getCheckElement())
      .append($filterElement)
      .append($textElement);
    $element.append(removeButtonElement);
  }
```